plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ExpressionConverter.java

```diff
@@ -115,93 +111,97 @@ private static Expression toIcebergExpression(String columnName, Type type, Doma
-                Marker high = range.getHigh();
-                Marker.Bound lowBound = low.getBound();
-                Marker.Bound highBound = high.getBound();
-
-                if (lowBound == EXACTLY && highBound == EXACTLY) {
-                    // case ==
-                    if (getIcebergLiteralValue(type, low).equals(getIcebergLiteralValue(type, high))) {
-                        expression = or(expression, equal(columnName, getIcebergLiteralValue(type, low)));
-                    }
-                    else { // case between
-                        Expression between = and(
-                                greaterThanOrEqual(columnName, getIcebergLiteralValue(type, low)),
-                                lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        expression = or(expression, between);
-                    }
-                }
-                else {
-                    if (lowBound == EXACTLY && low.getValueBlock().isPresent()) {
-                        // case >=
-                        expression = or(expression, greaterThanOrEqual(columnName, getIcebergLiteralValue(type, low)));
-                    }
-                    else if (lowBound == ABOVE && low.getValueBlock().isPresent()) {
-                        // case >
-                        expression = or(expression, greaterThan(columnName, getIcebergLiteralValue(type, low)));
-                    }
-
-                    if (highBound == EXACTLY && high.getValueBlock().isPresent()) {
-                        // case <=
-                        if (low.getValueBlock().isPresent()) {
-                            expression = and(expression, lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                        else {
-                            expression = or(expression, lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                    }
-                    else if (highBound == BELOW && high.getValueBlock().isPresent()) {
-                        // case <
-                        if (low.getValueBlock().isPresent()) {
-                            expression = and(expression, lessThan(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                        else {
-                            expression = or(expression, lessThan(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                    }
-                }
+                expression = or(expression, toIcebergExpression(columnName, range));
             }
             return expression;
         }
 
         throw new VerifyException("Did not expect a domain value set other than SortedRangeSet but got " + domainValues.getClass().getSimpleName());
     }
 
-    private static Object getIcebergLiteralValue(Type type, Marker marker)
+    private static Expression toIcebergExpression(String columnName, Range range)
```

Comment:
Argh. Instead validating by eyeballing that previous logic matches current one I would very much prefer a unit test for that. Can do?

Reply:
seems like unit tests may be sufficient here. Outside of the PR scope, it'd be cool to have tests that assert splits generated by iceberg split manager based on filter.

Reply:
No, the new implementation is different and creates a different `Expression` tree.

I did not dare to maintain https://github.com/trinodb/trino/blob/03d8e5abc686c5c4a9f96ca14db07e2aed880174/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ExpressionConverter.java#L148-L152 logic which was too clever for me.

Reply:
> seems like unit tests may be sufficient here.

let me disagree.  Although I don't think i am an expert on Iceberg, I do believe I am _simplifying_ the code.
I do agree it would be better to have a test for this, but I do not see it as required in this PR.

Reply:
(discussed offline)